### PR TITLE
[tanshiyu1999] iP

### PR DESCRIPTION
**Pull request by tanshiyu1999**
# Project Duke
> My name is Bob.

What I can do
- List Tasks
- Add Tasks 
- Edit & Delete Task

Things you need to know (about me)
1. The best Duke Bot
1. My name is Bob
1. Smile More
1. I am ~Pikachu~ Bob

```
running in the 90s
```

- [x] #739
- [ ] https://github.com/octo-org/octo-repo/issues/740
- [ ] Add delight to the experience when all tasks are complete :tada:

This site was built using `Java` with guides from [CS2103T website](https://nus-cs2103-ay2324s1.github.io/).

### CODE REVIEW

File: src/main/java/duke/task/Task.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    /**
+     * Get the description of the class.
+     * @return string The string representation of the class description.
+     */
+    public String getDescription() {
+        return this.description;
+    }
+
+    /**
+     * This method prints a horizonyal line.
```
Comment (leezhanpeng) [inline src/main/java/duke/task/Task.java:47]: Just a small typo here.

File: src/main/java/duke/task/Task.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    /**
+     * Mark the duke.task as undone.
+     */
+    public String markUndone() {
+        String strToPrint = "OK, I've marked this duke.task as not done yet:\n";
+        this.isDone = false;
+        return strToPrint + ("[" + this.getStatusIcon() + "] " + this.getDescription());
+    }
```
Comment (leezhanpeng) [inline src/main/java/duke/task/Task.java:82]: Maybe adding a @return tag here is relevant as the return of the string may not come naturally for this function?

File: src/main/java/duke/task/Event.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        list.add(eventTask);
+        if (splitEvent[1].equals("1")) {
+            list.get(list.size() - 1).markDoneNoPrint();
+        }
+    }
+
+    /**
+     * Default System.out when this function is called.
+     * @return string information in the Event instance.
+     */
```
Comment (leezhanpeng) [inline src/main/java/duke/task/Event.java:45]: For @return tags in Javadoc, it is actually indicated to be optional for members like toString() as they are [self-explanatory](https://google.github.io/styleguide/javaguide.html#s7.3.1-javadoc-exception-self-explanatory), so there is no problem to give one-liners like this:
> /**
> &ensp;\* Returns the string information in the Event instance.
> &ensp;*/

There are some other files with the same case.

File: src/main/java/duke/taskList/TaskList.java [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+            return "An error occurred";
+        }
+        return strToPrint += ("Now you have " + list.size() + " tasks in the list.");
+    }
+
+
+    /**
+     * Adds a event into ArrayList.
+     * @param command The duke.command specifying what Event to add.
+     */
```
Comment (leezhanpeng) [inline src/main/java/duke/taskList/TaskList.java:120]: Again, a @return tag here might be useful?

Other instances of @return tag can be added to other functions in this file as well!

**Review by leezhanpeng — COMMENTED** [review]
LGTM. I didn't manage to spot any concerns, only some nitpicks. Great job!

File: src/main/java/duke/storage/Storage.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+     */
+    public Storage() {
+
+    }
+
+    /**
+     * Load the data from duke.txt into the TaskList inside of the program.
+     * @param list the TaskList that the data is loaded to from duke.txt.
+     */
+    public void load(TaskList list) {
```
Comment (quzhetao01) [inline src/main/java/duke/storage/Storage.java:29]: Can afford to be more specific here by specifying what you will be loading (Eg. loadTaskList)

File: src/main/java/duke/storage/Storage.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+            File theDir = new File("./src/main/data");
+            if (!theDir.exists()) {
+                theDir.mkdirs();
+            }
+            File myObj = new File("./src/main/data/duke.txt");
+            Scanner myReader = new Scanner(myObj);
+            while (myReader.hasNextLine()) {
+                String data = myReader.nextLine();
+                String type = data.substring(0, 1);
+                switch (type) {
```
Comment (quzhetao01) [inline src/main/java/duke/storage/Storage.java:42]: The `switch-case` statement needs to follow the coding standard stated in the course (no indentation for the case clause).

File: src/main/java/duke/taskList/TaskList.java [hunk 7]
```diff
@@ -0,0 +1,358 @@
...
+            e.printStackTrace();
+            return "An error occurred";
+        }
+        strToPrint += (eventTask) + "\n";
+        return strToPrint += ("Now you have " + list.size() + " tasks in the list.");
+    }
+
+
+    /**
+     * Add a Todo into the ArrayList.
```
Comment (quzhetao01) [inline src/main/java/duke/taskList/TaskList.java:193]: @return tag is missing here

File: src/main/java/duke/taskList/TaskList.java [hunk 8]
```diff
@@ -0,0 +1,358 @@
...
+                    if (i == 1) {
+                        throw new DukeException("There is no description");
+                    }
+                    if (i == splitCommand.length - 1) {
+                        throw new DukeException("Please insert the deadline");
+                    }
+                }
+            }
+
+            if (hasBy == false) {
```
Comment (quzhetao01) [inline src/main/java/duke/taskList/TaskList.java:261]: abit of nitpicking, but maybe we can shorten this to !hasBy

**Review by quzhetao01 — COMMENTED** [review]
Javadoc comments require some improvements. Naming-wise, mostly well done other than some nitpicking. Remember the switch-case coding standard for CS2103T course!

File: src/main/java/duke/Duke.java [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+    public String getResponse(String input) {
+        return "Duke heard: " + input;
+    }
+
+    public String getCommandResponse(String command) {
```
Comment (evanyan13) [inline src/main/java/duke/Duke.java:61]: Maybe can include JavaDoc for this method to differentiate from the `getResponse()` method above?

File: src/main/java/duke/taskList/TaskList.java [hunk 10]
```diff
@@ -0,0 +1,358 @@
...
+        Task deadlineTask;
+        try {
+            LocalDate d1 = LocalDate.parse(date);
+            deadlineTask = new Deadline(description, date, d1);
+        } catch (DateTimeParseException e) {
+            deadlineTask = new Deadline(description, date);;
+        }
+
+        try {
+            FileWriter myWriter = new FileWriter("./src/main/data/duke.txt", true);
```
Comment (evanyan13) [inline src/main/java/duke/taskList/TaskList.java:283]: Is it possible to store the address of the input/output file as a constant somewhere instead of hardcoding them every time? It might be easier for you to edit it as well if the file location changes afterwards

```suggestion
            FileWriter myWriter = new FileWriter(FILE_PATH, true);
```

**Review by evanyan13 — COMMENTED** [review]
LGTM! Only identified a few nits in my comments 👍
